templates/common/_base/files/etc-networkmanager-dispatcher.d-90-long-hostname.yaml

```diff
@@ -0,0 +1,40 @@
+filesystem: "root"
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/90-long-hostname"
+contents:
+  inline: |
+    #!/bin/bash
+    #
+    # On Google Compute Platform (GCP) the hostname may be too long (>63 chars).
+    # During firstboot the hostname is set in the initramfs before NetworkManager
+    # runs.
+    #
+    # This hook is a simple work around: if the host name is longer than 63
+    # characters, then the hostname is truncated at the _first_ dot.
+    #
+    # Additionally, this hook does not break DNS or cluster resolution,
+    # since NetworkManager sets the appropriate /etc/resolv.conf settings.
+
+
+    IF=$1
+    STATUS=$2
+
+    log() { logger --tag "network-manager/$(basename $0)" "${@}"; }
+
+    if [ "${HOSTAME}" == "localhost" ]; then
```

Comment:
is this supposed to be `!=`, or am I just misunderstanding?

Reply:
Yikes....good catch. Fixed and pushed. Retetsting that now.